### PR TITLE
evince: update to 46.3

### PR DESCRIPTION
**Pull request by ognevny**

### CODE REVIEW

**Comment by ognevny** [comment]
what is https://github.com/msys2/MINGW-packages/actions/runs/8801506958/job/24155022720?pr=20712#step:11:4056?

**Comment by iahung4** [comment]
Does this PR fix these issues?

https://github.com/msys2/MINGW-packages/issues/19723
https://github.com/msys2/MINGW-packages/issues/20537

p/s: I lost access to my @iahung3 account.

**Comment by ognevny** [comment]
> Does this PR fix these issues?
> 
> https://github.com/msys2/MINGW-packages/issues/19723
> https://github.com/msys2/MINGW-packages/issues/20537
> 
> p/s: I lost access to my @iahung3 account.

currently I can't check this. you can download the package from artifacts

**Comment by iahung4** [comment]
> > Does this PR fix these issues?
> > #19723
> > #20537
> > p/s: I lost access to my @iahung3 account.
> 
> currently I can't check this. you can download the package from artifacts

The issues are still there.

**Comment by iahung3** [comment]
@ognevny Can you fix these issues?

p/s: I'm back.

**Comment by ognevny** [comment]
> @ognevny Can you fix these issues?
> 
> p/s: I'm back.

I'm on vacation until the 5th on May, I don't have access to the pc, sorry

File: mingw-w64-evince/PKGBUILD [hunk 1]
```diff
@@ -16,6 +16,7 @@ msys2_references=(
 license=("spdx:GPL-2.0-or-later")
 depends=("${MINGW_PACKAGE_PREFIX}-glib2"
          "${MINGW_PACKAGE_PREFIX}-cairo"
+         "${MINGW_PACKAGE_PREFIX}-dbus"
```
Comment (lazka) [inline mingw-w64-evince/PKGBUILD:19]: Does it not build without dbus? I'd like to avoid it.

Comment (ognevny) [reply mingw-w64-evince/PKGBUILD]: `-Ddbus=false` disables it. done

**Comment by ognevny** [comment]
pushed an update to 46.3

**Comment by hmartinez82** [comment]
@iahung3 The missing icon is an upstream issue. I'll try to get them an icon soon.

**Comment by hmartinez82** [comment]
@iahung3  https://gitlab.gnome.org/GNOME/evince/-/merge_requests/701

**Comment by ognevny** [comment]
should I backport this change?

**Comment by hmartinez82** [comment]
That's up to you. It's just cosmetics at this point 😅. And it's not only text, you'd have to add the .ico to the patch too.

**Comment by ognevny** [comment]
> That's up to you. It's just cosmetics at this point 😅. And it's not only text, you'd also have to add the .ico file to the patch.

I guess the binary form will be installed in patch file. let's give it a try

**Comment by ognevny** [comment]
@hmartinez82 what does this mean?
```
FAILED: shell/shell_evince.rc_evince.o
"C:\msys64\clang64\bin/windres.EXE" "../evince-46.3/shell/evince.rc" "shell/shell_evince.rc_evince.o" "--preprocessor-arg=-MD" "--preprocessor-arg=-MQshell/shell_evince.rc_evince.o" "--preprocessor-arg=-MFshell/shell_evince.rc_evince.o.d"
llvm-rc: Error in ICON statement (ID Icon):
Incorrect icon/cursor Reserved field; should be 0.
```

**Comment by hmartinez82** [comment]
Oh I guess I should try with Clang, I only tried UCRT64

**Comment by hmartinez82** [comment]
@ognevny I just built it with the icon using the CLANG64 environment. Not only I don't see your error, it does create evince.exe with the icon correctly: 
image

**Comment by ognevny** [comment]
in recipe I copied the ico (I downloaded it separately) with `cp -a ${srcdir}/evince.ico shell`. is the path correct?

**Comment by ognevny** [comment]
in UCRT64 I get `C:\msys64\ucrt64\bin/windres.EXE: icon file `./../evince-46.3/shell/evince.ico' does not contain icon data`

**Comment by hmartinez82** [comment]
The new files should be
`shell/evince.rc`
`shell/evince.ico`
and there are changes to `shell/meson.build`

**Comment by ognevny** [comment]
> The new files should be `shell/evince.rc` `shell/evince.ico` and there are changes to `shell/meson.build`

I download the patch, but removed .ico file from there. I download it separately from `https://gitlab.gnome.org/GNOME/evince/-/blob/24967ecf7f658bb69e512d6e09287c1ccc02c11c/shell/evince.ico`

**Comment by hmartinez82** [comment]
> in UCRT64 I get `` C:\msys64\ucrt64\bin/windres.EXE: icon file `./../evince-46.3/shell/evince.ico' does not contain icon data ``

Then you are definitely missing a file. Look at the original PR. There are three things

**Comment by ognevny** [comment]
> > in UCRT64 I get `` C:\msys64\ucrt64\bin/windres.EXE: icon file `./../evince-46.3/shell/evince.ico' does not contain icon data ``
> 
> Then you are definitely missing a file. Look at the original PR. There are three things

it was a surprise for me, but downloading with that url from `source` is incorrect. it works after I download the with `download` button 🤦‍♂️

File: mingw-w64-evince/evince.ico
Comment (hmartinez82) [inline mingw-w64-evince/evince.ico:1]: This file is corrupted. It's not a valid ICO file.

Comment (ognevny) [reply mingw-w64-evince/evince.ico:1]: why?

Comment (ognevny) [reply mingw-w64-evince/evince.ico:1]: oh `warning: in the working copy of 'mingw-w64-evince/evince.ico', CRLF will be replaced by LF the next time Git touches it`

Comment (hmartinez82) [reply mingw-w64-evince/evince.ico:1]: I just downloaded this file and it's corrupted. That's all the information I have.

**Comment by ognevny** [comment]
I guess I can't push it

**Comment by hmartinez82** [comment]
I'd say go ahead and merge this, then I'll take a look into the icon

**Comment by hmartinez82** [comment]
@ognevny I created this PR into this PR's branch branch https://github.com/ognevny/MINGW-packages/pull/5

**Comment by ognevny** [comment]
> @ognevny I created this PR into this PR's branch branch https://github.com/ognevny/MINGW-packages/pull/5

what `inline=false` do?

**Comment by hmartinez82** [comment]
> > @ognevny I created this PR into this PR's branch branch [ognevny#5](https://github.com/ognevny/MINGW-packages/pull/5)
> 
> what `inline=false` do?

I didn't ask 😅 That's the link to download the file from the Merge Request, untouched. I did not create that link myself. That url is the unmodified url you get for downloading:
image

**Comment by ognevny** [comment]
> > > @ognevny I created this PR into this PR's branch branch [ognevny#5](https://github.com/ognevny/MINGW-packages/pull/5)
> > 
> > what `inline=false` do?
> 
> I didn't ask 😅 That's the link to download the file from the Merge Request, untouched. I did not create that link myself. That url is the unmodified url you get for downloading:
> image
> 

understood...

**Comment by hmartinez82** [comment]
@lazka This should be good to go. I'll take a look at the console in a separate PR. This branch is already getting old 😅

**Comment by ognevny** [comment]
@hmartinez82 thanks a lot for help!

**Pull request merged**